Adding utils.format_quantity

rkschamer opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind feature

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

PR adds a `utils.format_quantity`, which converts a decimal number into a canonical kubernetes quantity. 

There is already `utils.parse_quantity` which I used in my project, but I also needed a way to convert a `Decimal` number back into a Kubernetes quantity. This PR proposes to add this functionality to the Python K8s client.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #2205

#### Special notes for your reviewer:

I couldn't find any tests for the `utils` modules, neither the `parse_quantity` function. I'd be very happy to add some tests for `format_quantity`, but so far I just didn't find the right place for them. Please let me know where I should put them.

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Adding `utils.format_quantity` to convert decimal numbers into a canonical Kubernetes quantity.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

> I think that's not applicable for this change - I'm not sure though.

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

rkschamer commented:
/kind feature

roycaihw reviewed (commented):
> I couldn't find any tests for the utils modules, neither the parse_quantity function. I'd be very happy to add some tests for format_quantity, but so far I just didn't find the right place for them. Please let me know where I should put them.

You could use https://github.com/kubernetes-client/python/blob/master/kubernetes/e2e_test/test_utils.py

rkschamer commented:
> > I couldn't find any tests for the utils modules, neither the parse_quantity function. I'd be very happy to add some tests for format_quantity, but so far I just didn't find the right place for them. Please let me know where I should put them.
> 
> You could use https://github.com/kubernetes-client/python/blob/master/kubernetes/e2e_test/test_utils.py

Perfect, thank you. I've now added the tests for `format_quantity` and also took the liberty to add a test for `parse_quantity`. Since pytest haven't been used in `test_utils.py` I refrained from using [`@pytest.mark.parametrize`](https://docs.pytest.org/en/6.2.x/parametrize.html). Please let me know if I should change this.

With the tests, I now consider my change as ready for review and hence, I've also changed to an actual PR.

rkschamer commented:
Hi @roycaihw,

I don't want to be pushy, but do you think, you can have a look on the PR? 

Thank you,
René

rkschamer commented:
/remove-lifecycle stale

rkschamer commented:
I appreciate any review on the PR.

/remove-lifecycle stale

rkschamer commented:
Dear reviewers, can you help to review this PR?

Much appreciated.

/assign @kubernetes/sig-pr-reviews
/remove-lifecycle rotten
/help

yliaog commented:
thanks for the PR

/lgtm
/approve

green-digger commented:
@rkschamer hi, i think the function is missing from the __init__ so prevents the import being

 `from kubernetes.utils import format_quantity`
 
 instead it is
 
 `
from kubernetes.utils.quantity import format_quantity,
 `
